### PR TITLE
PERF: asv for import

### LINKED ISSUE #26663: PERF: asv for import time
Not entirely sure how to do it with asv, but at the command line its just `python3 -X importtime -c "import pandas"`

- [ ] import time with `python3 -X importtime -c "import pandas"`
- [ ] import size with something like `import pandas, psutil; psutil.Process().memory_info()`

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #26663

for py37 this uses `-X importtime` to get a more precise number without subprocess overhead.  Doesn't actually do anything with it, but what the heck.

Couple of unrelated cleanups.

### CODE REVIEW

File: asv_bench/benchmarks/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import subprocess
+import sys
+
+from pandas.compat import PY37
+
+
+class TimeImport:
+    def time_import(self):
+        if PY37:
+            cmd = [sys.executable, "-X", "importtime", "-c", "import pandas as pd"]
```
Comment (jreback) [inline asv_bench/benchmarks/package.py]: is there a reason you are not using ``subprocess.run(...).stdout`` ? much simpler

**Review by jreback — CHANGES_REQUESTED** [review]

File: asv_bench/benchmarks/package.py (hunk 1 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/package.py]: force of habit, will change

Comment (jbrockmendel) [reply asv_bench/benchmarks/package.py]: updated

**Review by TomAugspurger — APPROVED** [review]

File: asv_bench/benchmarks/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            p = subprocess.run(cmd, stderr=subprocess.PIPE)
```
Comment (jreback) [inline asv_bench/benchmarks/package.py]: can you add a comment for this as 3.7 actually does the timing, where for < 3.7 asv is doing the timing.

File: asv_bench/benchmarks/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+            line = p.stderr.splitlines()[-1]
+            field = line.split(b"|")[-2].strip()
+            total = int(field)  # microseconds
+            return total
+
+        cmd = [sys.executable, "-c", "import pandas as pd"]
+        p = subprocess.run(cmd, stderr=subprocess.PIPE)
```
Comment (jreback) [inline asv_bench/benchmarks/package.py]: is this a lint error to assign to p?

**Review by jreback — APPROVED** [review]
very minor comment, merge on green.

File: asv_bench/benchmarks/package.py (hunk 3 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/package.py]: i think it should be, no idea why it isn't.  will fix anyway

File: asv_bench/benchmarks/package.py (hunk 2 above)
Comment (jbrockmendel) [reply asv_bench/benchmarks/package.py]: comment added + green

**Comment by WillAyd** [comment]
Thanks @jbrockmendel

**Pull request merged**